RELEASE.md

```diff
@@ -0,0 +1,36 @@
+# Releasing connect-go
+
+This document outlines how to create a release of connect-go.
+
+1. Clone the repo, ensuring you have the latest main.
+
+2. On a new branch, open [connect.go](connect.go) and change the `Version` constant. Do not just remove the `-dev` suffix: look at the release history and the unreleased commits to choose a new semantic version number.
+
+   ```patch
+   -const Version = "1.14.0-dev"
+   +const Version = "1.14.0"
+   ```
+
+4. Open a PR titled "Prepare for vX.Y.Z" ([Example PR #661](https://github.com/connectrpc/connect-go/pull/661)). Once it's reviewed and CI passes, merge it. *Make sure no new commits are merged until the release is complete.*
```

Comment:
Need to emphasize this a bit more. Adding it to its own line at least. Could consider using an [important callout note](https://github.com/orgs/community/discussions/16925)

```suggestion
4. Open a PR titled "Prepare for vX.Y.Z" ([Example PR #661](https://github.com/connectrpc/connect-go/pull/661)). Once it's reviewed and CI passes, merge it.

   *Make sure no new commits are merged until the release is complete.*
```

Reply:
Important note can't be indented into the list, so have chosen to go with the separate paragraph.